File: 532-k-diff-pairs-in-an-array/k-diff-pairs-in-an-array.py
class Solution:
    def findPairs(self, nums: List[int], k: int) -> int:
        nums.sort()
        find = set()
        n = len(nums)
        arr = []
        for i in range(n):
            value = nums[i] - k
            if value in find:
                arr.append([nums[i], value])
            find.add(nums[i])
        set_arr = set()
        for pair in arr:
            set_arr.add(tuple(pair))
        return len(set_arr)

File: 532-k-diff-pairs-in-an-array/k-diff-pairs-in-an-array.java
class Solution {
    public int findPairs(int[] nums, int k) {
        Arrays.sort(nums);
        Set<Integer> find = new HashSet<>();
        List<List<Integer>> arr = new ArrayList<>();

        for (int i = 0; i < nums.length; i++) {
            int value = nums[i] - k;
            if (find.contains(value)) {
                arr.add(Arrays.asList(nums[i], value));
            }
            find.add(nums[i]);
        }

        Set<List<Integer>> setArr = new HashSet<>();
        for (List<Integer> pair : arr) {
            setArr.add(pair);
        }

        return setArr.size();
    }
}